### ISSUE TITLE
layer quant_conv2d_58 expects 2 weights, but the saved weights have 1 elements

### ISSUE THREAD

**Issue by KienPM** [larq/larq#377]
### Describe the bug
I use Keras to save, then load weights. I got this error
```
ValueError: Layer #144 (named "quant_conv2d_58" in the current model) was found to correspond to layer quant_conv2d_58 in the save file. However the new layer quant_conv2d_58 expects 2 weights, but the saved weights have 1 elements.
```

### Expected behavior
Am I missing any config when load weights?

### Environment
Ubuntu 18.04
Python 3.6.9
TensorFlow version: tensorflow-gpu 1.15.0
Larq version: larq 0.8.2

**Comment by lgeiger** [comment]
Could you provide a code example to reproduce this issue?

**Comment by KienPM** [comment]
this is my code to save weights
```
checkpoint = ModelCheckpoint(log_dir + 'ep{epoch:03d}-loss{loss:.3f}-val_loss{val_loss:.3f}.h5',
                                 monitor='val_loss', save_weights_only=True, save_best_only=True, period=1)
```
and this is my code to load weights
```
self.yolo_model = yolo_body(Input(shape=(None, None, 3)), num_anchors // 3, num_classes)
self.yolo_model.load_weights(self.model_path)  # make sure model, anchors and classes match
```

**Comment by leonoverweel** [comment]
Hey @KienPM, I wasn't able to reproduce your issue. Here's a code snippet that uses (hopefully) as much stuff you're using as I could think of, including a `QuantConv2D` layer that appears in the error message you posted originally. In the snippet, I create the model, fit it, save it, and then reload its weights.

```python
import tensorflow as tf
import larq as lq


def create_model():
    model = tf.keras.Sequential(
        [
            lq.layers.QuantConv2D(
                128,
                3,
                kernel_quantizer="ste_sign",
                kernel_constraint="weight_clip",
                use_bias=False,
                input_shape=(28, 28, 1),
            ),
            tf.keras.layers.Flatten(),
            tf.keras.layers.Dense(10, activation="softmax"),
        ]
    )
    model.compile(
        optimizer="adam", loss="sparse_categorical_crossentropy", metrics=["accuracy"]
    )

    return model


# Simple Dataset
num_samples = 1000
((train_x, train_y), (test_x, test_y),) = tf.keras.datasets.mnist.load_data()

train_y = train_y[:num_samples]
test_y = test_y[:num_samples]

train_x = train_x[:num_samples].reshape(-1, 28, 28, 1) / 255.0
test_x = test_x[:num_samples].reshape(-1, 28, 28, 1) / 255.0

# Checkpoint callback
save_location = "outputs/337-model.h5"
cp_callback = tf.keras.callbacks.ModelCheckpoint(
    filepath=save_location, save_weights_only=True, verbose=1
)

# Fit and save model
model = create_model()
model.fit(
    train_x,
    train_y,
    epochs=5,
    validation_data=(test_x, test_y),
    callbacks=[cp_callback],
)
print(f"\nModel trained and saved in `{save_location}`.\n")

# Load and evaluate model
print("Reloading and evaluating model...")

model = create_model()
model.load_weights(save_location)

model.evaluate(test_x, test_y, verbose=2)
```

Here's the output:

```
>>> python 337.py
[... tensorflow warnings / information omitted ...]
Train on 1000 samples, validate on 1000 samples
[... epochs 1 - 4 omitted ...]
Epoch 5/5
 992/1000 [============================>.] - ETA: 0s - loss: 0.2965 - accuracy: 0.9526
Epoch 00005: saving model to outputs/337-model.h5
1000/1000 [==============================] - 0s 393us/sample - loss: 0.2941 - accuracy: 0.9530 - val_loss: 3.6289 - val_accuracy: 0.7830

Model trained and saved in `outputs/337-model.h5`.

Reloading and evaluating model...
1000/1 - 0s - loss: 4.2733 - accuracy: 0.7830
```

As you can see from the evaluation accuracy matching the last training validation accuracy, the model gets reloaded correctly. Can you edit this snippet with more of your own code (in a way that I can run it without having the rest of your code) so we can figure out where the problem is?

**Comment by KienPM** [comment]
@leonoverweel @lgeiger 
This is my whole project, I base on this project https://github.com/qqwweee/keras-yolo3.
I added some files (yolo3/bnn_model.py, bnn_train.py, bnn_yolo.py, bnn_yolo_video.py which use your QuantConv2D instead of Keras' QuantConv2D.
model_data/ep102-loss36730.007-val_loss3322.113.h5 is my trained weights. You can reproduce the error when run bnn_yolo_video.py

I hope that it can help we figure out where the problem is.

https://drive.google.com/open?id=11nNLpmx8lcf2gAUt7mNjWqc88Ee8XP3J

**Comment by leonoverweel** [comment]
Hey, it's a bit hard to diagnose in a large codebase I'm not familiar with. Can you try to reduce the issue to something we can run from a single file?

One thing that comes to mind: how did you train the `model_data/ep102-loss36730.007-val_loss3322.113.h5` weights? If they were trained with Keras' `QuantConv2D` instead of Larq's `QuantConv2D`, that could be a potential pitfall.

**Comment by KienPM** [comment]
Thank you for your response!
The model is create at bnn_model.py, and trained from bnn_train.py
You can reproduce the error when run bnn_yolo_video.py
And yes, I'm missing ```use_bias=False```, may be it cause the problem. Let me try again!

**Comment by leonoverweel** [comment]
Sounds good, let me know if this works 👍

**Comment by KienPM** [comment]
> Sounds good, let me know if this works

It works after add ```use_bias=False```, thank you so much!

**Comment by leonoverweel** [comment]
Great! I'll close this issue for now; if you run into any other problems, feel free to reopen.
